core/include/vecmem/memory/details/device_address_space.hpp

```diff
@@ -0,0 +1,14 @@
+/* VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+namespace vecmem {
+
+/// Custom definition for the device memory adress space
+enum class device_address_space { global = 0, local = 1 };
```

Comment:
It might make more sense to have this file outside of the `details/` namespace in the end. I was probably a bit too eager with putting it in here. :thinking:

Reply:
👍